client/components/backup-retention-management/index.tsx

```diff
@@ -70,53 +73,82 @@ const BackupRetentionManagement: FunctionComponent = () => {
 	) as number;
-	const estimatedCurrentSiteSize = useEstimatedCurrentSiteSize();
 
+	const lastBackupSize = useSelector( ( state ) =>
+		getBackupCurrentSiteSize( state, siteId )
+	) as number;
+
+	const estimatedCurrentSiteSize = lastBackupSize * ( STORAGE_ESTIMATION_ADDITIONAL_BUFFER + 1 );
 	const currentSiteSizeText = useStorageText( estimatedCurrentSiteSize );
 	const storageLimitText = useStorageText( storageLimitBytes );
 
-	const retentionOptionsCards: Record< number, RetentionRadioOptionType > = {
-		[ RETENTION_OPTIONS.RETENTION_DAYS_7 ]: usePrepareRetentionOptions(
-			translate( '7 days' ),
-			RETENTION_OPTIONS.RETENTION_DAYS_7,
-			currentRetentionPlan,
-			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_7
-		),
-		[ RETENTION_OPTIONS.RETENTION_DAYS_30 ]: usePrepareRetentionOptions(
-			translate( '30 days' ),
-			RETENTION_OPTIONS.RETENTION_DAYS_30,
-			currentRetentionPlan,
-			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_30
-		),
-		[ RETENTION_OPTIONS.RETENTION_DAYS_120 ]: usePrepareRetentionOptions(
-			translate( '120 days' ),
-			RETENTION_OPTIONS.RETENTION_DAYS_120,
-			currentRetentionPlan,
-			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_120
-		),
-		[ RETENTION_OPTIONS.RETENTION_DAYS_365 ]: usePrepareRetentionOptions(
-			translate( '1 year' ),
-			RETENTION_OPTIONS.RETENTION_DAYS_365,
-			currentRetentionPlan,
-			retentionSelected === RETENTION_OPTIONS.RETENTION_DAYS_365
-		),
-	};
+	const getOptionsInitialState = (): RetentionOptionInput[] => [
+		{
+			id: 7,
+			spaceNeededInBytes: 0,
+			upgradeRequired: false,
+		},
+		{
+			id: 30,
+			spaceNeededInBytes: 0,
+			upgradeRequired: false,
+		},
+		{
+			id: 120,
+			spaceNeededInBytes: 0,
+			upgradeRequired: false,
+		},
+		{
+			id: 365,
+			spaceNeededInBytes: 0,
+			upgradeRequired: false,
+		},
+	];
+
+	const [ retentionOptionsCards, setRetentionOptionsCards ] = useState( getOptionsInitialState );
+
+	// Update the retention options cards with the correct values dynamically
+	useEffect( () => {
+		if ( isFetching ) {
+			return;
+		}
+
+		setRetentionOptionsCards( ( previousOptions ) => {
+			const newOptions = previousOptions.map( ( option ) => {
+				const spaceNeededInBytes = estimatedCurrentSiteSize * option.id;
+				const upgradeRequired = spaceNeededInBytes > storageLimitBytes;
+
+				return {
+					...option,
+					spaceNeededInBytes,
+					upgradeRequired,
+				};
+			} );
+
+			return newOptions;
+		} );
+	}, [ estimatedCurrentSiteSize, isFetching, storageLimitBytes ] );
 
 	const updateRetentionRequestStatus = useSelector( ( state ) =>
 		getBackupRetentionUpdateRequestStatus( state, siteId )
 	);
 
 	// Set the retention period selected when the user selects a new option
-	const onRetentionSelectionChange = useCallback( ( value: number ) => {
-		if ( value ) {
-			setRetentionSelected( value );
-		}
-	}, [] );
+	const onRetentionSelectionChange = useCallback(
+		( value: number ) => {
+			if ( value ) {
```

Comment:
Double-checking noticed that we added this check because we were setting the default selection to 0. Fixing it to set a valid option as default.

Reply:
It appears that just adding a valid option as default is not too straightforward and, for timely reasons, I think it would be better to address this improvement in another PR. I commit to it.

Reply:
Sounds good, thanks 👍